pages/index.tsx

```diff
@@ -1,12 +1,42 @@
-import * as React from 'react'
+import React, { useState, useEffect } from 'react'
 import type { NextPage } from 'next'
 import Container from '@mui/material/Container'
 import Typography from '@mui/material/Typography'
 import Box from '@mui/material/Box'
-import Copyright from '../src/Copyright'
+import Grid from '@mui/material/Grid'
+import CircularProgress from '@mui/material/CircularProgress'
+import ClassCard from '../src/components/ClassCard'
 import Link from '../src/Link'
 
+interface ClassData {
+	number: string
+	aliases: string
+	department: string
+	deprecated: boolean
+	foundational: string
+	name: string
+	link: string
+	course_id: string
+}
+
 const Home: NextPage = () => {
+	const [loading, setLoading] = useState<boolean>()
+	const [classes, setClasses] = useState<[ClassData]>()
+
+	useEffect(() => {
+		setLoading(true)
+		fetch('https://omshub-readonly.gigalixirapp.com/classes')
```

Comment:
Per legal discussion, we should redact this data for now, it's still not clear what the legality is of using the OMSCentral data. Let's either TODO this block, or use mock data instead.